### ISSUE TITLE
SpriteComponents with texture doesn't respect size

### ISSUE THREAD

**Issue by naithar** [bevyengine/bevy#411]
**Platform:**
macOS 10.15.4 / iOS 13.6

**Description:**
Using `SpriteComponents` with texture and specific size value doesn't change the way sprite is rendered. 
Using only color for `ColorMaterial` renders sprite correctly.
Using `scale` works for both texture and color, but color requires size being specified.

**Example:**

Texture:
```
let texture_handle = asset_server.load("icon.png").unwrap();

commands
.spawn(Camera2dComponents::default())
.spawn(SpriteComponents {
  material: materials.add(ColorMaterial::modulated_texture(texture_handle.into(), Color::rgb(0.2, 0.2, 0.8))),
  sprite: Sprite { size: Vec2::new(50.0, 50.0) },
  ..Default::default()
});
```

Color:
```
commands
.spawn(Camera2dComponents::default())
.spawn(SpriteComponents {
  material: materials.add(Color::rgb(0.2, 0.2, 0.8).into()),
  sprite: Sprite { size: Vec2::new(50.0, 50.0) },
  ..Default::default()
});
```
<img height="200" alt="Снимок экрана 2020-09-01 в 14 24 42" src="https://user-images.githubusercontent.com/3750083/91843796-051c2400-ec5f-11ea-8443-1e89fe8b9868.png"> <img height="200" alt="Снимок экрана 2020-09-01 в 14 25 28" src="https://user-images.githubusercontent.com/3750083/91843798-06e5e780-ec5f-11ea-9c15-bd6fa80edaa6.png"> <img height="200" src="https://user-images.githubusercontent.com/3750083/91844783-ba031080-ec60-11ea-998b-3430d2ad403b.png" /> <img height="200" src="https://user-images.githubusercontent.com/3750083/91844789-bd969780-ec60-11ea-8142-c82315bdd8aa.png" />

**Comment by TheNeikos** [comment]
This is because of this system: https://github.com/bevyengine/bevy/blob/57177c9e980e19a8a377ef8c45d34d6a5f68c22c/crates/bevy_sprite/src/sprite.rs#L21-L34

**Comment by naithar** [comment]
@TheNeikos well, I guess then the question is - is this an expected behavior. 
I would expect it to fire only if sprite size is not set or zero (but this also might be the value developer want for some reason), but I might be wrong :)

**Comment by TheNeikos** [comment]
I would even go as far as that only the sprite size is important and that the dev should always fill it.

**Comment by naithar** [comment]
> This is because of this system:

That's the result when I remove this system, os I guess something should be done with the way this system is handled:
<img height="300" src="https://user-images.githubusercontent.com/3750083/91887425-72e24300-ec93-11ea-801f-1b4108b0e0b7.png" />

**Comment by naithar** [comment]
Maybe there is a way to save current behavior as default one, but also fixing it when developer specifies an actual size? Like using some trait like `SpriteSize`?

**Comment by TheNeikos** [comment]
I think the size could be made into an enum:

```rust
enum Size {
  Manual(f32, f32),
  Computed(f32, f32),
  Undefined,
}
```

This way the system seen here (https://github.com/bevyengine/bevy/issues/411#issuecomment-684908109) will only assign a new size when it is Computed or Undefined making the intent clear. It would also default to `Size::Undefined` in case one doesn't want to manually size it.

**Comment by cart** [comment]
Closed by #430
